### PR TITLE
feat: Add Home Screen Quick actions

### PR DESCRIPTION
**Pull request by charlella**

### CODE REVIEW

File: Mail/Views/QuickAction.swift [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import Foundation
+import UIKit
+
+enum QuickActionType: String {
+    case newMessage = "NewMessage"
+    case search = "Search"
+    case support = "Support"
+}
+
+enum QuickAction: Equatable {
```
Comment (valentinperignon) [inline Mail/Views/QuickAction.swift:28]: I would prefer a name like HomeScreenShortcut

File: Mail/Views/SplitView.swift [hunk 2]
```diff
@@ -269,6 +271,26 @@ struct SplitView: View {
         }
     }
 
+    private func handleApplicationShortcut(_ notification: Publishers.ReceiveOn<NotificationCenter.Publisher, DispatchQueue>
+        .Output) {
+        guard let shortcut = notification.object as? UIApplicationShortcutItem else { return }
+        guard let quickAction = QuickAction(shortcutItem: shortcut) else { return }
```
Comment (valentinperignon) [inline Mail/Views/SplitView.swift:277]: Only one guard

File: Mail/Views/SplitView.swift [hunk 3]
```diff
@@ -269,6 +271,26 @@ struct SplitView: View {
...
+        case .newMessage:
+            mainViewState.composeMessageIntent = .new(originMailboxManager: mailboxManager)
+            matomo.track(eventWithCategory: .homeScreenShortcuts, name: "newMessage")
+        case .search:
+            mainViewState.isShowingSearch = true
+            matomo.track(eventWithCategory: .homeScreenShortcuts, name: "search")
+        case .support:
+            openURL(URLConstants.chatbot.url)
+            matomo.track(eventWithCategory: .homeScreenShortcuts, name: "support")
+        }
```
Comment (valentinperignon) [inline Mail/Views/SplitView.swift:291]:
```suggestion
        }
        
        @InjectService var matomo: MatomoUtils
        matomo.track(eventWithCategory: .homeScreenShortcuts, name: quickAction.rawValue)
```

File: MailCore/Utils/Notification+Name.swift [hunk 4]
```diff
@@ -25,4 +25,5 @@ public extension Notification.Name {
     static let updateComposeMessageBody = Notification.Name(rawValue: "updateComposeMessageBody")
     static let openNotificationSettings = Notification.Name(rawValue: "openNotificationSettings")
     static let printNotification = Notification.Name(rawValue: "printNotification")
+    static let userPerformedShortcut = Notification.Name(rawValue: "userPerformedShortcut")
```
Comment (valentinperignon) [inline MailCore/Utils/Notification+Name.swift:28]:
```suggestion
    static let userPerformedHomeScreenShortcut = Notification.Name(rawValue: "userPerformedHomeScreenShortcut")
```

File: Mail/Views/SplitView.swift (hunk 3 above)
Comment (charlella) [reply Mail/Views/SplitView.swift:291]: replace quickAction with homeScreenShortcut

**Review by valentinperignon — APPROVED** [review]

**Pull request merged**